### PR TITLE
[AutoParallel] add collate_fn for dist_loader

### PR DESCRIPTION
**Pull request by zhaoyinglia**
### PR types

Function optimization
### PR changes

Others
### Describe

- (1) add collate_fn for dist_loader to generate mini-batch data by merging the sample list
    - 解决问题:
        - dataset 返回各种数据类型都可以正确组batch，与框架支持的数据类型相同
    - 目前限制：
        - 只支持半自动，所有数据batch维统一切分大小。
        - 如果dataset返回带有batch信息的数据，则返回的数据类型必须是 array，因为需要用到 numpy.split 来对数据进行切分。
- (2) add lr_scheduler to update learning rate after every training step
    - 目前限制：
        - 仅支持如果用户定义lr_scheduler, 则每个训练step都会更新学习率，用户暂不能控制更新频率

### CODE REVIEW

**Review by Caozhou1995 — DISMISSED** [review]

**Review by JZ-LIANG — APPROVED** [review]
LGTM
